src/ibek/ioc.py

```diff
@@ -67,24 +68,25 @@ def make_entity_class(definition: Definition, support: Support) -> Type[Entity]:
     for arg in definition.args:
         # make_dataclass can cope with string types, so cast them here rather
         # than lookup
-        arg_type = cast(type, arg.type)
         metadata: Any = None
+        arg_type: type
         if isinstance(arg, ObjectArg):
 
-            def make_conversion(name: str = arg.type) -> Conversion:
-                module_name, entity_name = name.split(".", maxsplit=1)
-                entity_cls = getattr(getattr(modules, module_name), entity_name)
+            def lookup_instance(id__):
+                try:
+                    return id_to_entity[id__]
+                except KeyError:
+                    raise ValidationError(f"{id__} is not in {list(id_to_entity)}")
```

Comment:
We can back out the `id__` name now

```suggestion
            def lookup_instance(id):
                try:
                    return id_to_entity[id]
                except KeyError:
                    raise ValidationError(f"{id} is not in {list(id_to_entity)}")
```